src/legacy/ui/public/directives/field_name/field_name_icon.tsx

```diff
@@ -17,62 +17,47 @@
  * under the License.
  */
 import React from 'react';
+import { palettes, EuiIcon } from '@elastic/eui';
+import { IconSize } from '@elastic/eui/src/components/icon/icon';
 
+interface IconMapEntry {
+  icon: string;
+  color: string;
+}
 interface Props {
   type: string;
```

Comment:
Hi could we potentially make this:

```ts
type: 'boolean' | 'conflict' | 'date' | ... | string;
```

this will effectively be the same type, due to the `string` in the end, but would help autocomplete that type for the actual known string values.

Reply:
for sure!

Reply:
done